x-pack/plugins/monitoring/server/routes/api/v1/_health/errors_helpers/errors_query.ts

```diff
@@ -5,24 +5,26 @@
  * 2.0.
  */
 
-import type { MetricbeatMonitoredProduct, QueryOptions } from '../types';
+import type { MetricbeatMonitoredProduct, PackagesMonitoredProduct, QueryOptions } from '../types';
 
 const MAX_BUCKET_SIZE = 50;
 
 /**
  * Returns a nested aggregation of error messages per event.datasets.
  * Each module (beats, kibana...) can contain one or multiple metricsets with error messages
  */
-interface MetricbeatErrorsQueryOptions extends QueryOptions {
-  products: MetricbeatMonitoredProduct[];
+interface ErrorsQueryOptions extends QueryOptions {
+  products: MetricbeatMonitoredProduct[] | PackagesMonitoredProduct[];
+  errorQueryType: 'metricbeatErrorsQuery' | 'packageErrorsQuery';
 }
 
-export const metricbeatErrorsQuery = ({
+export const errorsQuery = ({
   timeRange,
   timeout,
   products,
-}: MetricbeatErrorsQueryOptions) => {
-  if (!timeRange) throw new Error('metricbeatErrorsQuery: missing timeRange parameter');
+  errorQueryType,
+}: ErrorsQueryOptions) => {
+  if (!timeRange) throw new Error(`${errorQueryType}: missing timeRange parameter`);
```

Comment:
I'm wondering if `event.module` and `metricset.name` are produced out of the box by a standalone integration (ie that does not rely on metricbeat) or if it's a product of using metricbeat under the hood ? If the latter, this will break when/if we update the integration to stop relying on metricbeat. This should be fine if we have coverage to catch this, but maybe there are fields populated regardless of the integration implementation that we could rely on ?

Reply:
I can see them when I installed the elasticsearch integration. I needed to add the `metricset.name` in the integrations ([PR](https://github.com/elastic/integrations/pull/4973)) but apart from that it worked when I tested it locally  
<details>
 <summary>The query I used in the Dev Tools</summary>
  GET metrics-elasticsearch.stack_monitoring.node-default/_search
{
  "query":{
    "bool": {
      "filter":{
        "bool": {
           "must": [
                {
                  "exists": {
                    "field": "error.message"
                  }
                },
                {
                  "terms": {
                    "event.module":["elasticsearch"]
                  }
                }
            ]
        }
      }
    }
  },
  "aggs": {
    "errors_aggregation": {
      "terms": {
        "field": "event.module"
      },
      "aggs": {
        "errors_by_dataset":{
          "terms": {
            "field": "metricset.name"
          }, 
          "aggs": {
            "latest_docs": {
              "top_hits": {
                "sort": [
                  {
                    "@timestamp": {
                      "order": "desc"
                    }
                  }
                ],
                "size": 100,
                "_source": {
                  "includes": ["@timestamp", "error", "metricset"]
                }
              }
            }
          }    
        }
      }
    }
  }
}

</details>
I started with adding an integration test, but I faced some issues on the way so I will add it to a follow-up PR on a separate branch - https://github.com/jennypavlova/kibana/commit/e2f9af27502aae5b97d2d0c7c844f95eb85e689a


Do you have any suggestions on how to check a specific integration and which index should I use for the case you are mentioning so I can verify that the fields are there?

Reply:
[This WIP documentation](https://github.com/elastic/observability-dev/blob/c265598d5d31768412c3fb6ba89023326556cc7f/docs/infra-obs-ui/stack-monitoring_integration-packages.md#metrics) could clarify my question. I was wondering if `metricset.name` was created by the metricbeat subprocess and if we could rely on a field that's populated by the agent instead (maybe `data_stream.dataset`) so the query will continue to work if we remove metricbeat process from the integrations.
My concern is if the fields don't exist anymore and the query never returns anything because of it, it will probably go unnoticed. Let's ignore it for this change but we should log this for future reference

Reply:
As we discussed I changed `metricset.name` to `data_stream.dataset` and the `event.module` is changed to `service.type`